File: packages/ui-flutter/lib/src/drawer.dart
import 'package:flutter/widgets.dart';

import 'dialog.dart';
import 'sheet.dart';

/// Mirrors `packages/ui/src/components/drawer.tsx` (a vaul bottom drawer).
/// Functionally identical to [KinetixSheet] here — a thin pass-through,
/// the same call the other ports made.
class KinetixDrawer extends StatelessWidget {
  const KinetixDrawer({
    super.key,
    required this.visible,
    required this.onDismiss,
    required this.child,
  });

  final bool visible;
  final VoidCallback onDismiss;
  final Widget child;

  @override
  Widget build(BuildContext context) {
    return KinetixSheet(visible: visible, onDismiss: onDismiss, child: child);
  }
}

typedef KinetixDrawerHeader = KinetixDialogHeader;
typedef KinetixDrawerFooter = KinetixDialogFooter;
typedef KinetixDrawerTitle = KinetixDialogTitle;
typedef KinetixDrawerDescription = KinetixDialogDescription;

File: packages/ui-swiftui/Sources/KinetixUI/Drawer.swift
//
// Drawer.swift — KinetixDrawer.
//
// Mirrors packages/ui/src/components/drawer.tsx (a vaul bottom drawer
// with a drag handle). Functionally identical to KinetixSheet here, so
// it's a thin pass-through — the same call the Compose port made
// (`KinetixDrawer` → `KinetixSheet`). `shouldScaleBackground` has no
// SwiftUI equivalent and is dropped.
//

import SwiftUI

public struct KinetixDrawer<Content: View>: View {
    @Binding private var isPresented: Bool
    private let content: Content

    public init(isPresented: Binding<Bool>, @ViewBuilder content: () -> Content) {
        self._isPresented = isPresented
        self.content = content()
    }

    public var body: some View {
        KinetixSheet(isPresented: $isPresented) { content }
    }
}

public typealias KinetixDrawerHeader<Content: View> = KinetixDialogHeader<Content>
public typealias KinetixDrawerFooter<Content: View> = KinetixDialogFooter<Content>
public typealias KinetixDrawerTitle = KinetixDialogTitle
public typealias KinetixDrawerDescription = KinetixDialogDescription
